fc/parsing/actions.py

```diff
@@ -934,32 +946,66 @@ def _add_input_variables(self):
                 # Add the new input variable, with ontology annotation
                 # TODO: Extract this into a helper method that DeclareVariable processing etc can also use
                 name = self.model._get_unique_name('protocol__' + var.local_name)  # TODO: Make method public
-                cmeta_id = self.model.get_unique_cmeta_id('protocol__' + var.local_name)
                 units = self.units.get_unit(var.units)
-                variable = self.model.add_variable(name, units, cmeta_id=cmeta_id)
-                self.model.rdf.add((variable.rdf_term, PRED_IS, var.rdf_term))
-            # TODO: Set initial_value on variable if given?
+                variable = self.model.add_variable(name, units)
+                self.model.add_cmeta_id(variable)
+                self.model.rdf.add((variable.rdf_identity, PRED_IS, var.rdf_term))
+
+            # Store initial value if given
+            if variable is not None and var.initial_value is not None:
+                self.initial_values[var.rdf_term] = var.initial_value
 
     def _add_or_replace_equations(self):
-        """Process define statements and modify the model's equations accordingly."""
+        """
+        Process define statements and modify the model's equations accordingly. Also sets initial values determined in
+        input statements.
+        """
+        # Apply all modifications (sympy_equations contains _only_ equations from define statements)
         for eq in self.sympy_equations:
             lhs = eq.lhs
             if lhs.is_Derivative:
                 var = lhs.args[0]
-                if var.initial_value is None:
+                term = self.model.get_ontology_terms_by_variable(var)[0]
```

Comment:
This isn't guaranteed to work. Instead I'd do a pre-processing phase before you iterate through equations, where instead of a map from RDF term to initial value you create a map from variable to initial value, looking up the terms. Then you can look up directly in that map here.

Reply:
Otherwise this looks good!